modules/kubernetes/linkerd/main.tf

```diff
@@ -0,0 +1,205 @@
+    }
+
+    annotations = {
+      "linkerd.io/inject" = "disabled"
+    }
+
+    name = "linkerd-cni"
+  }
+}
+
+# Create self signed certificate for linkerd-trust-anchor: https://linkerd.io/2.10/tasks/automatically-rotating-control-plane-tls-credentials/
+resource "tls_private_key" "linkerd_trust_anchor" {
+  algorithm   = "ECDSA"
+  ecdsa_curve = "P256" # P256 required by linkerd: https://github.com/linkerd/linkerd2/blob/b9aa32f9b20057c7166347825428e53525962b9c/pkg/issuercerts/issuercerts.go#L144-L146
+}
+
+# More information regarding creating the trust anchor: https://linkerd.io/2.10/tasks/automatically-rotating-control-plane-tls-credentials/#cert-manager-as-an-on-cluster-ca
+resource "tls_self_signed_cert" "linkerd_trust_anchor" {
+  key_algorithm         = tls_private_key.linkerd_trust_anchor.algorithm
+  private_key_pem       = tls_private_key.linkerd_trust_anchor.private_key_pem
+  validity_period_hours = 87600
+  early_renewal_hours   = 78840
+  is_ca_certificate     = true
+
+  subject {
+    common_name = "root.linkerd.cluster.local"
+  }
+
+  allowed_uses = [
+    "cert_signing",
+    "crl_signing",
+  ]
+}
+
+resource "kubernetes_secret" "linkerd_trust_anchor" {
+  metadata {
+    name      = "linkerd-trust-anchor"
+    namespace = kubernetes_namespace.this.metadata[0].name
+  }
+
+  data = {
+    "tls.crt" = tls_self_signed_cert.linkerd_trust_anchor.cert_pem
+    "tls.key" = tls_private_key.linkerd_trust_anchor.private_key_pem
+  }
+
+  type = "kubernetes.io/tls"
+}
+
+# Create self signed certificate for webhook-issuer-tls: https://linkerd.io/2.10/tasks/automatically-rotating-webhook-tls-credentials/
+resource "tls_private_key" "webhook_issuer_tls" {
+  algorithm   = "ECDSA"
+  ecdsa_curve = "P256" # P256 required by linkerd: https://github.com/linkerd/linkerd2/blob/b9aa32f9b20057c7166347825428e53525962b9c/pkg/issuercerts/issuercerts.go#L144-L146
+}
+
+# More information regarding the webhook issuer: https://linkerd.io/2.10/tasks/automatically-rotating-webhook-tls-credentials/#save-the-signing-key-pair-as-a-secret
+resource "tls_self_signed_cert" "webhook_issuer_tls" {
+  key_algorithm         = tls_private_key.webhook_issuer_tls.algorithm
+  private_key_pem       = tls_private_key.webhook_issuer_tls.private_key_pem
+  validity_period_hours = 87600
+  early_renewal_hours   = 78840
+  is_ca_certificate     = true
+
+  subject {
+    common_name = "webhook.linkerd.cluster.local"
+  }
+
+  allowed_uses = [
+    "cert_signing",
+    "crl_signing",
+  ]
+}
+
+resource "kubernetes_secret" "webhook_issuer_tls" {
+  metadata {
+    name      = "webhook-issuer-tls"
+    namespace = kubernetes_namespace.this.metadata[0].name
+  }
+
+  data = {
+    "tls.crt" = tls_self_signed_cert.webhook_issuer_tls.cert_pem
+    "tls.key" = tls_private_key.webhook_issuer_tls.private_key_pem
+  }
+
+  type = "kubernetes.io/tls"
+}
+
+# Install linkerd-cni helm chart
+locals {
+  values_cni = templatefile("${path.module}/templates/values-cni.yaml.tpl", {})
```

Comment:
Can we inline the values instead of using locals.

Reply:
Done!